### DISCUSSION TITLE
Saving data to csv

### DISCUSSION THREAD

**Discussion by asifsultan-THD** [pybamm-team/PyBaMM#1861] [Q&A] [answered]
I was following your first video to store the data in the csv file. I am not able to read the csv file. I imported pandas and then trying to go open but  it was showing me this error
"UnicodeDecodeError: 'utf-8' codec can't decode byte 0x80 in position 0: invalid start byte"

**Comment by actualpriyanshu** [comment]
If you are using pandas, you could read csv by simply doing
```
import pandas as pd
data = pd.read_csv(filename, encoding= 'unicode_escape')
```

**Reply by asifsultan-THD** [reply]
Thanks Priyanshu. It worked but on seeeing the csv file , it is coming something like this without any numerical data

data.head()

ò	@wý_
0	@	NaN
1	Pmöû	NaN
2	@CÚ«¨_ñ	NaN
3	@uÀi4è	NaN
4	@LÄË¿Pß	NaN

**Comment by actualpriyanshu** [comment, marked as answer]
The default behaviour of `save_data` is to create a pickle file. You need to pass the `to_format` parameter to `save_data`.

For example
```
solution.save_data("filename.csv", ["Time [s]", "Terminal voltage [V]"], to_format="csv")
```
